pandas/core/groupby.py

```diff
@@ -1139,6 +1184,49 @@ def _try_cast(self, result, obj):
 
         return result
 
+    def _cython_transform(self, how, numeric_only=True):
```

Comment:
It does a couple different things (looking up the cython func, creating `accum` which is an array where running-totals are stored) but a lot is duplicated.  I'll try to factor out the common stuff.